### DISCUSSION TITLE
No benchmarks for zstd?

### DISCUSSION THREAD

**Discussion by dudicoco** [lz4/lz4#1548] [General]
I've been comparing lz4 with zstd and saw a significant advantage with using zstd.
I'm wondering if there's any configuration option that would make lz4 outperform zstd?

```sh
curl -L https://www.kernel.org/pub/linux/kernel/v5.x/linux-5.15.8.tar.gz | tar -xzf -
rm -f archive.tar
tar -cf archive.tar ./linux-5.15.8/

rm -f archive.tar.lz4
time lz4 archive.tar archive.tar.lz4
du -h archive.tar.lz4

rm -f archive.tar.zstd
time zstd archive.tar -o archive.tar.zstd
du -h archive.tar.zstd

rm -f archive-lz4.tar
time lz4 -d archive.tar.lz4 archive-lz4.tar

rm -f archive-zstd.tar
time zstd -d archive.tar.zstd -o archive-zstd.tar
```

```sh
Compressed 1137315840 bytes into 332336507 bytes ==> 29.22%                    
lz4 archive.tar archive.tar.lz4  1.75s user 0.12s system 99% cpu 1.896 total
324M    archive.tar.lz4

archive.tar          : 16.44%   (  1.06 GiB =>    178 MiB, archive.tar.zstd)   
zstd archive.tar -o archive.tar.zstd  2.36s user 0.15s system 108% cpu 2.315 total
187M    archive.tar.zstd

archive.tar.lz4      : decoded 1137315840 bytes                                
lz4 -d archive.tar.lz4 archive-lz4.tar  0.54s user 0.32s system 97% cpu 0.881 total

archive.tar.zstd    : 1137315840 bytes                                         
zstd -d archive.tar.zstd -o archive-zstd.tar  0.80s user 0.15s system 120% cpu 0.786 total

```

**Comment by Cyan4973** [comment]
> saw a significant advantage with using zstd.

Do you mean in term of decompression speed ?

Both `zstd` and `lz4` use async i/o for decompression, with a single thread for the actual decompression operation.
I would expect both to feature similar speed performance, because i/o tends to be the bottleneck.

That being said, in situation where i/o is a significant bottleneck, having less data to read becomes a measurable advantage, and in these cases `zstd` would win.

Unfortunately, there is nothing `lz4` can do to beat `zstd` on compression ratio. At best you can increase the compression level, this will decrease the difference, at the cost of compression speed. But `zstd` will remain the stronger option.

**Reply by dudicoco** [reply]
I mean in both compression and decompression.

zstd compressed archive in my tests is about half the size of the lz4 archive and it took both programs around the same time to compress.

Both programs took about the same time to decompress while zstd decompressed an archive almost twice as big as lz4's archive.
